### PR TITLE
feat: Adds addProcessor(Processor) method in SpoonAPI.

### LINKED ISSUE #300: add addProcessor(Processor) in SpoonAPI
see http://lists.gforge.inria.fr/pipermail/spoon-discuss/2015-August/001794.html

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #300

### CODE REVIEW

File: src/main/java/spoon/Launcher.java [hunk 1]
```diff
@@ -88,6 +90,7 @@
 	protected JSAPResult jsapActualArgs;
 
 	private List<String> processors = new ArrayList<String>();
+	private List<Processor<? extends CtElement>> processorsInstance = new ArrayList<Processor<? extends CtElement>>();
```
Comment (boeserwolf91) [inline src/main/java/spoon/Launcher.java]: I do NOT like the name. Why not just call it "processors" without the word "Instance"?

File: src/main/java/spoon/Launcher.java [hunk 2]
```diff
@@ -485,6 +493,14 @@ protected void processArguments() {
 	}
 
 	/**
+	 * Gets the list of processors instance to be initially applied during the
+	 * processing.
+	 */
+	protected List<Processor<? extends CtElement>> getProcessorsInstance() {
```
Comment (boeserwolf91) [inline src/main/java/spoon/Launcher.java]: Why not call it getProcessors()?...
Is the access modifier with protected the right one? You could make it public and either return a unmodifiable list or an array.

File: src/main/java/spoon/Launcher.java (hunk 1 above)
Comment (GerardPaligot) [inline src/main/java/spoon/Launcher.java]: Me too but `processors` is declared above this line.

File: src/main/java/spoon/Launcher.java (hunk 2 above)
Comment (GerardPaligot) [inline src/main/java/spoon/Launcher.java]: `getProcessors()` already exists and these methods aren't exposed in the API.

File: src/main/java/spoon/Launcher.java (hunk 1 above)
Comment (boeserwolf91) [inline src/main/java/spoon/Launcher.java]: oh I see. I'm blind :smile: 
Anyway the name is horrible and sounds wrong. "processorInstances" would be the right way to go. But it's not much better.
Why not making it clearer in some way? Call processors -> processorNames and the processorInstances -> processors (the names aren't real processors, just the names)

File: src/main/java/spoon/Launcher.java (hunk 2 above)
Comment (boeserwolf91) [inline src/main/java/spoon/Launcher.java]: I can't see a getProcessors method. "just" getProcessorTypes()

File: src/main/java/spoon/Launcher.java (hunk 1 above)
Comment (boeserwolf91) [inline src/main/java/spoon/Launcher.java]: oh call the processors field processorTypes to math the getter method instead of processorNames

Comment (GerardPaligot) [inline src/main/java/spoon/Launcher.java]: Good idea!

File: src/main/java/spoon/Launcher.java (hunk 2 above)
Comment (GerardPaligot) [inline src/main/java/spoon/Launcher.java]: Oh yes, that's `getProcessorTypes()`. I change it!

Comment (boeserwolf91) [inline src/main/java/spoon/Launcher.java]: great. thank you. it looks better :+1:

File: src/main/java/spoon/SpoonAPI.java [hunk 3]
```diff
@@ -22,7 +24,12 @@
 	 * Adds a processor (fully qualified name).
 	 */
 	public void addProcessor(String name);
-	
+
+	/**
+	 * Adds a instance of a processor.
```
Comment (monperrus) [inline src/main/java/spoon/SpoonAPI.java]: "an instance' the user is responsible for keeping a pointer to it for later retrieving some processing information

**Pull request merged**